### PR TITLE
dnf5-group: add page

### PR DESCRIPTION
**Pull request by sohang3112**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page(s) follow the [style guide](/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):** `dnf5 version 5.2.7.0`

### CODE REVIEW

File: pages/linux/dnf-group.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# dnf group
+
+> Manage groups of related DNF packages on Fedora-based systems.
```
Comment (cyqsimon) [inline pages/linux/dnf-group.md:3]: Personally, I like how it's explained in the dnf man page:

> Groups  are  virtual  collections of packages. DNF keeps track of groups that the user selected ("marked") installed and can manipulate the comprising packages with simple commands.

And if I'm being scholastic here, "DNF packages" really isn't a thing; rather it's all just "RPM packages". But I think it's unnecessarily pedantic for this page anyways.

So how about this?

```suggestion
> Manage virtual collections of packages on Fedora-based systems.
```

File: pages/linux/dnf-group.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+`dnf group install {{groupname}}`
+
+- Remove DNF group:
+
+`dnf group remove {{groupname}}`
+
+- Upgrade DNF group:
+
+`dnf group upgrade{{groupname}}`
```
Comment (cyqsimon) [inline pages/linux/dnf-group.md:24]: Missed a space here?

**Review by spageektti — APPROVED** [review]

File: pages/linux/dnf-group.md (hunk 2 above)
Comment (sohang3112) [reply pages/linux/dnf-group.md]: image

@cyqsimon There's already a space there (checked in this PR's Files Changed tab), please recheck - the command is showing with space only:

`dnf group upgrade {{groupname}}`

File: pages/linux/dnf-group.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# dnf group
+
+> Manage virtual collections of packages on Fedora-based systems.
+> More information: <https://manned.org/man/dnf-group>.
```
[the remark this answers is not in the document]
Comment (sohang3112) [reply pages/linux/dnf-group.md:4]: AFAIK `dnf` and `dnf5` have exactly same interface, and at least on my Fedora 41 system `dnf` is aliased to `dnf5`. If this is still an issue, file can be modified to instead document `dnf5 group`, but IMO that shouldn't be needed (that documentation would be harder for user to find).

[the remark this answers is not in the document]
Comment (sohang3112) [reply pages/linux/dnf-group.md]: I have renamed command to `dnf5 group` - please check now.

[the remark this answers is not in the document]
Comment (sohang3112) [reply pages/linux/dnf-group.md]: > Maybe add an alias page dnf group that redirects to dnf5 group?

Done

**Comment by cyqsimon** [comment]
It's worth noting that dnf v5 only ***just*** became the default on Fedora 41. There are still many derivative distros out there (RHEL being the prime example) still on dnf v4, and due to their stability guarantees this will continue to be the case for a longgg time into the future.

For those unaware, there are quite a number of breaking changes from v4 to v5. Two in particular that are pertinent to the CLI are:

1. Some compound word long options (e.g. `--nobest`) are now hyphen-separated (`--no-best`).
2. Some deprecated commands (e.g. `dnf grouplist`) have been removed (superceded by `dnf group list`).

In short, I do not think it is a great idea to state that these two are equivalent at this moment in time. At minimum, there should be a notice/warning about how this may not be the case.

**Review by cyqsimon — CHANGES_REQUESTED** [review]
As explained in my comment.

**Comment by sohang3112** [comment]
@cyqsimon So what would be appropriate according to you - just delete *dnf-group.md* alias page and leave rest as it is? 

@sebastiaanspeck also what are your thoughts on this?

**Comment by cyqsimon** [comment]
With a bit more in-depth consideration, it's not as straight-forward as I thought. Sorry about making it sound easier than it actually is.

Here's a table to summarise the current situation. The strike-through ones are EOL:

| Distro                | dnf v4 | dnf v5                              |
|-----------------------|--------|-------------------------------------|
| RHEL 8 & 9            | `dnf`  | `microdnf`                          |
| ~Fedora 36 & lower~   | `dnf`  | `microdnf`                          |
| ~Fedora 37 & 38 & 39~ | `dnf`  | `dnf5`                              |
| Fedora 40             | `dnf`  | `dnf5`                              |
| Fedora 41 & higher    | N/A    | `dnf` (symlinked to `dnf5`), `dnf5` |

On how `microdnf` became `dnf5` and other related details, see [Fedora Wiki](https://fedoraproject.org/wiki/Changes/MajorUpgradeOfMicrodnf).

Also related: #12316. The situation is slightly different but still very relevant.

**Comment by cyqsimon** [comment]
> I tested with dnf 4 also in docker (`docker run -it fedora:40 /bin/bash`) - all the documented commands worked, only slight difference in `dnf group list` that `dnf5` had tabular output while dnf 4 didn't.

That's reassuring to hear. Knowing this, I feel less worried about using an alias page.

> Since `dnf` either points to dnf 4 or 5, how about simply making identical pages for `dnf group` and `dnf5 group` without alias, just copying whole content with command names changed?

In light of your verification, this would be redundant. Under the current situation, I think it's best we view the install name `dnf5` as a temporary name during the transition period, i.e. the "aliaser", with a note on potential incompatibilities on different distros. Maybe something like this?

> Note: between Fedora 37 and 40 (inclusive), DNF v4 is installed as `dnf` while DNF v5 is installed as `dnf5`.

What I suspect eventually will happen is Fedora 44 or something installs v5 as `dnf` directly and gets rid of the install name `dnf5`. If we do it like this, it will be easier to either deprecate or remove the page(s) for `dnf5` when that happens.

Apologies for the back and forth and all the hassle. Different install names is always a PITA to deal with.

**Comment by sohang3112** [comment]
@cyqsimon Now I made `dnf5 group` an alias of `dnf group` instead of other way around - please check it now.

File: pages/linux/dnf5-group.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# dnf5 group
+
+> This command is an alias of `dnf group`.
```
Comment (cyqsimon) [inline pages/linux/dnf5-group.md:3]: Would you mind me adding this line?

```suggestion
> This command is an alias of `dnf group`.
> Note: from Fedora 37 to 40 (inclusive), `dnf` runs DNF v4 while `dnf5` runs DNF v5.
```

File: pages/linux/dnf.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # dnf
 
 > Package management utility for RHEL, Fedora, and CentOS (replaces yum).
+> Some subcommands such as `config-manager` have their own usage documentation.
```
Comment (cyqsimon) [inline pages/linux/dnf.md:4]: And I recall that you had this previously?

```suggestion
> Some subcommands such as `group` and `config-manager` have their own usage documentation.
```

**Review by cyqsimon — APPROVED** [review]

File: pages/linux/dnf5-group.md (hunk 4 above)
Comment (sohang3112) [reply pages/linux/dnf5-group.md:3]: Sure that looks ok

File: pages/linux/dnf.md (hunk 5 above)
Comment (sohang3112) [reply pages/linux/dnf.md:4]: Yeah I'll put that back in

**Comment by sohang3112** [comment]
@cyqsimon Changes you suggested are done, please merge now
